### 1. General info of dataset GSE235063

This is the Jupyter Notebook for dataset GSE235063. Its dataset includes barcodes/genes/matrix files for each sample.

Thus, we need to simply incorparate these barcodes/genes/matrix files and generate an AnnData object for each sample. 

In total, there are 75 acute myeloid leukemia (AML) samples. The datasets includes both raw and processed information (thus, a total of 150 samples).

<span style="color:green">**[DX]**</span> samples from diagnosis

<span style="color:green">**[REM]**</span> samples from remission

<span style="color:green">**[REL]**</span> samples from relapse 

In [1]:
# Environment setup
import numpy as np
import pandas as pd
import scanpy as sc
import anndata as anndata
import scipy
import os

### 2. AnnData object of each sample

<span style="color:red">**IMPORTANT:**</span> rename files to get rid of prefixes

1. `barcodes.tsv`: cell barcodes, which go into `.obs`
2. `genes.tsv`: gene names, `.var`
3. `matrix.mtx`: the expression matrix, `.X`

<span style="color:red">**FAIL TO WORK ON PROCESSED SAMPLES????**</span>

In [13]:
general_input_path = '/scratch/user/s4543064/Xiaohan_Summer_Research/data/GSE235063'
general_output_path = '/scratch/user/s4543064/Xiaohan_Summer_Research/write/GSE235063'
count = 0

for sample in os.listdir(general_input_path):
    print("I am currently processing: ", sample)
    if 'processed' not in sample:
        sample_dir = os.path.join(general_input_path, sample)
        output_dir = os.path.join(general_output_path, sample) + '.h5ad'

        anndata = sc.read_10x_mtx(
            sample_dir,
            var_names='gene_symbols',  
            cache=False
        )
        print(anndata)
        
        # save the anndata object
        anndata.write_h5ad(output_dir, compression="gzip")

    else:
        count += 1
        print('Skipped samples:', str(count))

I am currently processing:  GSM7494323_AML19_REL_processed
Skipped samples: 1
I am currently processing:  GSM7494299_AML4_REL_raw
AnnData object with n_obs × n_vars = 6794880 × 33538
    var: 'gene_ids'
I am currently processing:  GSM7494294_AML10_REM_processed
Skipped samples: 2
I am currently processing:  GSM7494278_AML20_REM_processed
Skipped samples: 3
I am currently processing:  GSM7494302_AML22_REM_processed
Skipped samples: 4
I am currently processing:  GSM7494298_AML4_DX_processed
Skipped samples: 5
I am currently processing:  GSM7494281_AML5_REM_processed
Skipped samples: 6
I am currently processing:  GSM7494298_AML4_DX_raw
AnnData object with n_obs × n_vars = 6794880 × 33538
    var: 'gene_ids'
I am currently processing:  GSM7494322_AML19_DX_processed
Skipped samples: 7
I am currently processing:  GSM7494272_AML7_REL_raw
AnnData object with n_obs × n_vars = 6794880 × 33538
    var: 'gene_ids'
I am currently processing:  GSM7494305_AML21_REM_processed
Skipped samples: 8
I am c

Somehow the dimensions of all created AnnData objects are:

`AnnData object with n_obs × n_vars = 6794880 × 335381`